pipeline/nissl_to_ccfv3.py

```diff
@@ -0,0 +1,223 @@
+    resulting slice and the corresponding slice of the reference volume is
+    applied. (To make moving and reference more similar)
+    The registration is computed from the center of the volume to the two
+    extremities.
+
+    Parameters
+    ----------
+    reference_volume
+        Reference volume.
+    moving_volume
+        Moving volume of the same shape as the moving_volume.
+    nissl_volume
+        Nissl volume to register. It has to have same shape as reference_volume
+        and moving_volume and be from the same coordinate system as the moving
+        volume.
+
+    Returns
+    -------
+    warped_volume
+        Moving volume after the registration transformation.
+    nissl_warped
+        Nissl volume once the registration transformation are applied.
+    """
+    warped_volume = np.zeros_like(moving_volume)
+    nissl_warped = np.zeros_like(nissl_volume)
+    total_n_iterations = moving_volume.shape[0]
+    n_iterations = int(total_n_iterations / 2)
+
+    fixed = moving_volume[n_iterations, :, :]
+    for i in range(n_iterations):  # 0 -> 263
+        index = n_iterations - (i + 1)
+        warped = moving_volume[index]
+        fixed_ref = reference_volume[index]
+        nissl_slice = nissl_volume[index]
+
+        warped, nissl_slice = slice_registration(fixed, warped, nissl_slice)
+        warped, nissl_slice = slice_registration(fixed_ref, warped, nissl_slice)
+
+        fixed = warped
+        warped_volume[index] = warped
+        nissl_warped[index] = nissl_slice
+
+        if i % 5 == 0:
+            logger.info(f"{i + 1} / {total_n_iterations} registrations done")
+
+    fixed = moving_volume[n_iterations - 1]
+    for i in range(n_iterations - 1, total_n_iterations - 1):
+        warped = moving_volume[i + 1]
+        fixed_ref = reference_volume[i + 1]
+        nissl_slice = nissl_volume[i + 1]
+
+        warped, nissl_slice = slice_registration(fixed, warped, nissl_slice)
+        warped, nissl_slice = slice_registration(fixed_ref, warped, nissl_slice)
+
+        fixed = warped
+        warped_volume[i + 1] = warped  # 264 -> 527
+        nissl_warped[i + 1] = nissl_slice
+
+        if i % 5 == 0:
+            logger.info(f"{i + 1} / {total_n_iterations} registrations done")
+
+    return warped_volume, nissl_warped
+
+
+def main(
+    nissl_path: Path | str,
+    ccfv2_path: Path | str,
+    ccfv3_path: Path | str,
+    output_dir: Path | str | None = None,
+) -> int:
+    """Implement main function."""
+    logger.info("Loading volumes")
+    nissl = check_and_load(nissl_path)
+    ccfv2 = check_and_load(ccfv2_path)
+    ccfv3 = check_and_load(ccfv3_path)
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+
+    logger.info("Start registration...")
+    warped_atlas, warped_nissl = registration(ccfv3, ccfv2, nissl)
+
+    logger.info("Saving results...")
+    if output_dir is None:
+        output_dir = Path(__file__).parent / "nissl-to-ccfv3"
```

Comment:
Again, I feel like we should force the user to provide the output path

Reply:
Done in [154b7d3](https://github.com/BlueBrain/Deep-Atlas/pull/8/commits/154b7d34737f5c32b3bf6b2b612ccb73fcfaabb3).